pkg/deploy/controller/imagechange/controller.go

```diff
@@ -111,10 +111,10 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
 	}
 
 	if anyFailed {
-		return fatalError(fmt.Sprintf("couldn't update some deployment configs for trigger on image stream %q", imageapi.LabelForStream(stream)))
```

Comment:
If we returned a fatal error, the event would be lost, so the item wouldn't
be handled until the next sync window (which could be a minute or two).
For conflicts with the main controller, there is no reason not to update.

On Sun, Jul 10, 2016 at 10:45 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/deploy/controller/imagechange/controller.go
> <https://github.com/openshift/origin/pull/9578#discussion_r70196300>:
>
> > @@ -111,10 +111,10 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
> >  	}
> >
> >  	if anyFailed {
> > -		return fatalError(fmt.Sprintf("couldn't update some deployment configs for trigger on image stream %q", imageapi.LabelForStream(stream)))
>
> I thought returning a fatalError here just kept this error from causing an
> immediate requeue... I'm not seeing how it was preventing subsequent image
> change events from triggering attempts. Is the title accurate, or are we
> always wanting to retry immediately now?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9578/files/a850e5e780643ff932482d6a9224cbfbb137f332..ddd80e057bc3163119b64497ec1f0e1ba8b532fb#r70196300>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3MvUjc8Ah7a41qCC8XvJXCoC5yZks5qUa5LgaJpZM4I-xTE>
> .
>